### ISSUE TITLE
use_fsdp2_compile writes a dead adapter: torch.compile's _orig_mod. prefix, exit 0, all lora_B zero on reload

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#335]
Measured on 4xH100 (first time this path has ever been run — see #77).

`training.use_fsdp2_compile: true` with an FSDP preset trains, exits **0**, and writes an adapter whose tensors **are** trained (max|lora_B| 7.0e-3) — but every key carries torch.compile's wrapper prefix:

```
_orig_mod.base_model.model...      # written under use_fsdp2_compile
base_model.model...                # written by the paired non-compile run
```

`PeftModel.from_pretrained` matches **none** of them, emits only `UserWarning: Found missing adapter keys` (all 192), and leaves `lora_B` at its zero init. Result: **0/96 non-zero lora_B** — the adapter is a no-op. Reproduced 3/3.

This is the project's known worst failure shape: a run that completes successfully and produces a file that silently does nothing. The paired non-compile run on the identical config writes a live adapter (96/96), which is the control.

### Fix shape
Strip the `_orig_mod.` prefix before saving (or unwrap `model._orig_mod` for the save), the same way the streaming wrapper's `state_dict()` delegation handles its own `.inner.` segment.

### Acceptance
- [ ] an adapter saved under `use_fsdp2_compile` reloads into a normal model with non-zero `lora_B`
- [ ] a test asserting that, with a control proving the assertion would catch a dropped adapter (this repo already has that pattern in `tests/test_v07201.py`)
- [ ] no `_orig_mod.` keys in the saved file

Record: `benchmarks/gate-h100-validation.md` STEP 20.

**Comment by MakazhanAlpamys** [comment]
Fixed in `966b78e`, CI green on that commit (run 31195470152).

`strip_compile_prefix` normalises the saved adapter's keys, and runs only when the config actually enabled compile — so an ordinary run's adapter file is not rewritten for nothing.

Try:
```bash
soup train --config soup.yaml --gpus 4 --fsdp full_shard   # training.use_fsdp2_compile: true
```

Two things the tests assert beyond the obvious rename, both because the alternative is the same bug wearing new clothes:

- **the trained values survive** the rewrite. A repair that renamed keys correctly while losing numbers would reload cleanly and still train nothing.
- **a normal adapter is untouched** (returns 0), and **a missing adapter file is not an error** — full fine-tuning writes none, and that must not turn into a crash at the end of a completed run.

Implementation note worth keeping: the write goes through `mkstemp` + `os.replace`, because `load_file` memory-maps the file and writing over a live mapping fails on Windows with `os error 1224` — the same trap `adapter_fuse.py` already documents for an in-place `save_pretrained`. The test caught it on the first run.

Record: `benchmarks/gate-h100-validation.md` STEP 20.
